hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/chillmode/ChillModeHandler.java

```diff
@@ -88,16 +88,25 @@ public ChillModeHandler(Configuration configuration,
   @Override
   public void onMessage(ChillModeStatus chillModeStatus,
       EventPublisher publisher) {
-    try {
-      isInChillMode.set(chillModeStatus.getChillModeStatus());
-      scmClientProtocolServer.setChillModeStatus(isInChillMode.get());
-      scmBlockManager.setChillModeStatus(isInChillMode.get());
-      Thread.sleep(waitTime);
-    } catch (InterruptedException e) {
-      Thread.currentThread().interrupt();
-    } finally {
-      replicationManager.start();
+
+    isInChillMode.set(chillModeStatus.getChillModeStatus());
+    scmClientProtocolServer.setChillModeStatus(isInChillMode.get());
+    scmBlockManager.setChillModeStatus(isInChillMode.get());
+
+    if (!isInChillMode.get()) {
+      final Thread chillModeExitThread = new Thread(() -> {
+        try {
+          Thread.sleep(waitTime);
+        } catch (InterruptedException e) {
```

Comment:
Had offline discussion we shall make this daemon thread.

Reply:
Fixed.